rllib/algorithms/algorithm_config.py

```diff
@@ -473,7 +483,11 @@ class directly. Note that this arg can also be specified via
         if logger_creator is not None:
             self.logger_creator = logger_creator
 
-        return self.algo_class(
+        algo_class = self.algo_class
+        if isinstance(self.algo_class, str):
+            algo_class = get_algorithm_class(self.algo_class)
+
+        return algo_class(
```

Comment:
I feel like it would be a good place to always create a deepcopy and freezing it, right?